File: Punto4/recursivo.py
import time
import tracemalloc  

def factorial(n):
    if n == 0 or n == 1:
        return 1
    return n * factorial(n - 1)

num = 900


start_time = time.time()
tracemalloc.start()  

resultado = factorial(num)

current, peak = tracemalloc.get_traced_memory()  
tracemalloc.stop()  
end_time = time.time()

print(f"Factorial({num}) = {resultado}")
print(f"Tiempo de ejecución: {end_time - start_time:.6f} segundos")
print(f"Memoria usada: {peak / 1024:.2f} KB")

File: Punto4/recursivo.c
#include <stdio.h>
#include <time.h>
#include <sys/resource.h> 

unsigned long long factorial(int n) {
    if (n == 0 || n == 1)
        return 1;
    return n * factorial(n - 1);
}

int main() {
    int num = 900;
    struct rusage usage;
    
    clock_t start = clock(); 

    unsigned long long resultado = factorial(num);

    clock_t end = clock(); 
    getrusage(RUSAGE_SELF, &usage); 

    printf("Factorial(%d) = %llu\n", num, resultado);
    printf("Tiempo de ejecución: %f segundos\n", (double)(end - start) / CLOCKS_PER_SEC);
    printf("Memoria usada: %ld KB\n", usage.ru_maxrss);

    return 0;
}
